packages/web/src/utils/storage.ts

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2024 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export const safelyGetSessionStorage = (key: string): string | null => {
+  let value = null;
+  try {
+    value = window.sessionStorage.getItem(key);
```

Comment:
okay on the topic of session storage I don't have any strong feelings either way and it seems alright but [following MDN's docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage):

> survives over page reloads and restores.
Page reloads I think we can accept because they'll come up in spans. Restores seem like a rare case. So I think this is okayish?

> Duplicating a tab copies the tab's sessionStorage into the new tab.
Not sure about this one but it also seems rare enough?

I think we can just deploy this and check if it's good enough in practice.

Reply:
> survives over page reloads and restores

I did it intentionally that way. It is representing the browser tab rather than just the current window. I mean it can be adjusted if needed.

>  Duplicating a tab copies the tab's sessionStorage into the new tab.

I guess it is the same as the "restore" case. In the Cisco agent, we cover these use cases, but it requires a bigger refactor as the initialization would need to be asynchronous, and we would need to postpone span sending until the ID is assigned. It can be built on top of this as it is just an additional improvement.